offchain/host-runner/src/controller.rs

```diff
@@ -271,7 +271,7 @@ struct IdleState {
 }
 
 impl IdleState {
-    fn new(data: SharedStateData) -> Box<dyn State> {
+    fn create(data: SharedStateData) -> Box<dyn State> {
         Box::new(Self { data })
     }
 }
```

Comment:
To avoid using `create`, we can change this to:

```
impl IdleState {
    fn new(data: SharedStateData) -> Self {
        Self { data }
    }
}

impl From<IdleState> for Box<dyn State> {
    fn from(idle_state: IdleState) -> Self {
        Box::new(idle_state)
    }
}
```
And then we can do:
```
Some(IdleState::new(self.data).into())
```